python-wf-ortools-region-allocation/main.py

```diff
@@ -0,0 +1,293 @@
+        regions = result.json()
+        return regions
+
+    @app(app_id="region-allocation")
+    @step
+    def solve():
+        """Runs the model."""
+        pass  # Execution happens in sub-app.
+
+    @needs(predecessors=[solve, fetch_regions])
+    @step
+    def bundle_assets(result: dict, geojson_regions: dict):
+        """Enhances the result."""
+
+        # Colorize the regions based on the assignment and add metadata as a popup.
+        assignment_asset = copy.deepcopy(geojson_regions)
+        geojson_assignment(result, assignment_asset)
+
+        # Create choropleth map of the regions and their demand.
+        demand_asset = copy.deepcopy(geojson_regions)
+        geojson_demand(result, demand_asset)
+
+        # Create bar plot of the hub utilization
+        utilization_asset = create_utilization_plot(result)
+
+        # Write out the result
+        return nextmv.Output(
+            json_configurations={
+                "indent": None,
+                "separators": (",", ":"),
+            },
+            solution=result["solution"],
+            statistics=result["statistics"],
+            assets=[
+                nextmv.Asset(
+                    name="regions",
+                    content_type="json",
+                    visual=nextmv.Visual(
+                        visual_schema=nextmv.VisualSchema(value=nextmv.VisualSchema.GEOJSON),
+                        label="Assignment",
+                        visual_type="custom-tab",
+                    ),
+                    content=assignment_asset,
+                ),
+                nextmv.Asset(
+                    name="demand",
+                    content_type="json",
+                    visual=nextmv.Visual(
+                        visual_schema=nextmv.VisualSchema(value=nextmv.VisualSchema.GEOJSON),
+                        label="Demand",
+                        visual_type="custom-tab",
+                    ),
+                    content=demand_asset,
+                ),
+                nextmv.Asset(
+                    name="utilization",
+                    content_type="json",
+                    visual=nextmv.Visual(
+                        visual_schema=nextmv.VisualSchema(value=nextmv.VisualSchema.PLOTLY),
+                        label="Utilization",
+                        visual_type="custom-tab",
+                    ),
+                    content=utilization_asset,
+                ),
+            ],
+        )
+
+
+def main():
+    # Load input data
+    input = nextmv.load()
+
+    # Run workflow
+    flow = Flow("DecisionFlow", input.data)
+    flow.run()
+
+    # Write out the result
+    nextmv.write(flow.get_result(flow.bundle_assets))
+
+
+def geojson_assignment(
+    result: dict,
+    regions: dict,
+) -> None:
+    """Colorizes the regions based on the assignment and adds metadata as a popup."""
+    # Remove regions that are not in the solution
+    solution_regions = set(result["solution"]["assignments"].keys())
```

Comment:
Really a personal style nit, but I tend to use whitespace more. There is no whitespace at all in this whole function. After an `if`or `for` I normally use a newline, and also to separate logical chunks of code that  "go together"

Reply:
Yeah, I normally use a tad more too. This was written _fast_. Let me improve a little (maybe not fully reaching your style though :grimacing: ).

Reply:
Removed it and found an obsolete func that shouldn't have been in there anymore anyway. :sweat_smile: